sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1379,3 +1379,107 @@ case class GpuStringSplit(str: Expression, regex: Expression, limit: Expression)
+        //   willNotWorkOnGpu("str_to_map does not support regular expression delimiter(s)")
+        // }
+        if (str.numChars() == 0) {
+          willNotWorkOnGpu("delimiter is empty")
+        }
+      } else {
+        willNotWorkOnGpu("delimiter is null")
+      }
+    }
+  }
+
+  override def tagExprForGpu(): Unit = {
+    checkRegex(expr.pairDelim)
+    checkRegex(expr.keyValueDelim)
+  }
+
+  override def convertToGpu(str: Expression,
+                            pairDelim: Expression,
+                            keyValueDelim: Expression): GpuExpression =
+    GpuStringToMap(str, pairDelim, keyValueDelim)
+}
+
+case class GpuStringToMap(strs: Expression, pairDelim: Expression, keyValueDelim: Expression)
+    extends GpuExpression with ExpectsInputTypes with NullIntolerant {
+
+  def this(str: Expression, pairDelim: Expression) = {
+    this(str, pairDelim, GpuLiteral(":", StringType))
+  }
+
+  def this(str: Expression) = {
+    this(str, GpuLiteral(",", StringType), GpuLiteral(":", StringType))
+  }
+
+  override def dataType: MapType = MapType(StringType, StringType)
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, StringType, StringType)
+  override def prettyName: String = "str_to_map"
+  override def children: Seq[Expression] = Seq(strs, pairDelim, keyValueDelim)
+  override def nullable: Boolean = children.head.nullable
+  override def foldable: Boolean = children.forall(_.foldable)
+
+  override def columnarEval(batch: ColumnarBatch): Any = {
+    withResourceIfAllowed(strs.columnarEval(batch)) { val0 =>
+      withResourceIfAllowed(pairDelim.columnarEval(batch)) { val1 =>
+        withResourceIfAllowed(keyValueDelim.columnarEval(batch)) { val2 =>
+          (val0, val1, val2) match {
+            case (v0: GpuColumnVector, v1: GpuScalar, v2: GpuScalar) => toMap(v0, v1, v2)
+            case (v0: GpuScalar, v1: GpuScalar, v2: GpuScalar) =>
+              withResource(GpuColumnVector.from(v0, batch.numRows, v0.dataType)) { v0Col =>
+                toMap(v0Col, v1, v2)
+              }
+            case (v0, v1, v2) =>
+              throw new UnsupportedOperationException(s"Unsupported data '($v0: ${v0.getClass}," +
+                s" $v1: ${v1.getClass}, $v2: ${v2.getClass})' for GpuStringToMap.")
+          }
+        }
+      }
+    }
+  }
+
+  private def toMap(strs: GpuColumnVector,
+                    pairDelim: GpuScalar,
+                    keyValueDelim: GpuScalar): GpuColumnVector = {
+    // Firstly, split the input strings into lists of strings.
+    withResource(strs.getBase.stringSplitRecord(pairDelim.getBase)) { listsOfStrings =>
+      // Extract strings column from the output lists column.
+      withResource(listsOfStrings.getChildColumnView(0)) { stringsCol =>
+        // Split the key-value strings into pairs of strings of key-value (using maxSplit = 1).
+        // withResource(stringsCol.stringSplit(keyValueDelim.getBase, 1)) { keyValuePairs =>
```

Comment:
This will be enabled after cudf snapshot has been updated to accept the `maxSplit` parameter.